### DISCUSSION TITLE
Public file with download true does not trigger a download

### DISCUSSION THREAD

**Discussion by evrazi** [supabase/supabase#18438] [Questions]
Using this code

`
const { data } = supabase
  .storage
  .from('public-bucket')
  .getPublicUrl('folder/avatar1.png', {
    download: true,
  })
`

should trigger a download of a file in a public bucket as far as i understand the [docs](https://supabase.com/docs/reference/javascript/storage-from-getpublicurl?example=download-asset-in-public-bucket). Am i doing something wrong?

All I get from this is a public url.

**Comment by GaryAustin1** [comment]
Interesting as it has always been a helper function that just does string concatenation and does not make a server request like for signedURL.  

I don't see a fetch here, although it does look at the flag.  

https://github.com/supabase/storage-js/blob/fa44be8156295ba6320ffeff96bdf91016536a46/src/packages/StorageFileApi.ts#L509

**Reply by evrazi** [reply]
@GaryAustin1 I do agree on that and that part is working flawlessly. Hope I didn't misunderstand the docs then?

**Reply by GaryAustin1** [reply]
I'm thinking the docs are wrong.

**Reply by evrazi** [reply]
Thanks for clarifying. 

I hope the functionality will be added and not the docs updated, as it does save some time.

**Reply by yadaom** [reply]
Hello,
Is there a way to download the file with different name?
I storing the files with different name like asset id but want to download them with real time.
My file sizes are large, so I can't afford to read them in buffer and trigger the download. 

Thanks in advance.
Om
